### PR TITLE
Escape the keyword in stead of sanitizing. 

### PR DESCRIPTION
**Pull request by terw-dan**
## Summary

This PR can be summarized in the following changelog entry:
Escape the keyword rather than sanitizing it in the paper so the keyword can be matched correctly. 

## Relevant technical choices:

* In stead of stripping of certain characters that could interfere with regexes, they are now escaped. 

## Test instructions

This PR can be tested by following these steps:

* Try matching `$keyword`, this should now work. 

Fixes https://github.com/Yoast/wordpress-seo/issues/5244

### CODE REVIEW

File: js/values/Paper.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ var defaultAttributes = {
  * @returns {object} The attributes passed to the Paper.
  */
 var sanitizeAttributes = function( attributes ) {
-	attributes.keyword = sanitizeString( attributes.keyword );
+	attributes.keyword = escapeRegExp( attributes.keyword );
```
Comment (boblinthorst) [inline js/values/Paper.js]: Perhaps change method name if you change the action in the function is changed to escape instead of sanitize

**Comment by atimmer** [comment]
We shouldn't do the escaping when passing the value to the Paper. We should escape as late as possible before using the keyword in a regex. Because that will be less trivial, I am moving this to the 1.6 milestone.

**Comment by terw-dan** [comment]
Re-did this PR. In stead of escaping the keyword in the paper, we do it in the researches where necessary.

**Comment by boblinthorst** [comment]
CR: ok 👍

**Comment by jcomack** [comment]
Acceptance done 👍

**Pull request merged**